File: Ruby/SuppliesPackage.rb
#encoding: UTF-8

# Autores: Manuel Díaz-Meco y Ana Graciani Donaire
# 2º DGIIM		PDOO

# Esta clase representa a un paquete de suministros para una estación espacial.
# Puede contener armamento, combustible y/o energía para los escudos.

module Deepspace

	class SuppliesPackage

		# Constructor

		def initialize (ammo, fuel, shield)
			# Atributos de instancia (son privados)
			@ammoPower = ammo
			@fuelUnits = fuel
			@shieldPower = shield
		end

		# Método de copia (método de clase)

		def self.newCopy(s)
			new(s.ammoPower, s.fuelUnits, s.shieldPower)
		end

		public

		# Consultores públicos

		# Consultor del atributo ammoPower
		def ammoPower
			return @ammoPower
		end

		# Consultor del atributo fuelUnits
		def fuelUnits
			return @fuelUnits
		end

		# Consultor del atributo shieldPower
		def shieldPower
			return @shieldPower
		end
		
		def to_s
			return "Ammo Power: #{@ammoPower}, Fuel Units: #{@fuelUnits}, " \
				+ "Shield Power: #{@shieldPower}"
		end

	end

end

# Código de prueba

=begin
	paquete1 = Deepspace::SuppliesPackage.new(0,5.25,100.333)
	puts paquete1.inspect
	puts paquete1.ammoPower
	puts paquete1.fuelUnits
	puts paquete1.shieldPower
	paquete2 = Deepspace::SuppliesPackage.newCopy(paquete1)
	puts paquete2.inspect
	puts paquete2.ammoPower
	puts paquete2.fuelUnits
	puts paquete2.shieldPower
	puts paquete1.to_s
=end

File: Java/deepspace/src/deepspace/SuppliesPackage.java
/*
@authors Manuel Diaz-Meco Terrés y Ana Graciani Donaire
2ºDGIIM         PDOO


 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 * Esta clase representa a un paquete de suministros para una estación espacial
 * Puede contener armamento, combustible y/o energía para los escudos.
 */
class SuppliesPackage {         // Visibilidad de paquete, no ponemos nada

    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    // Constructor con visibilad de paquete que recibe como parámetro un
    // valor para cada uno de los atributos de la clase. 
    
    SuppliesPackage (float ammoPowergiven, float fuelUnitsgiven, float shieldPowergiven){
        
        ammoPower = ammoPowergiven;
        fuelUnits = fuelUnitsgiven;
        shieldPower = shieldPowergiven;
    }
    
    // Constructor copia con visibilidad de paquete.
    
    SuppliesPackage (SuppliesPackage s){
        
        ammoPower = s.ammoPower;
        fuelUnits = s.fuelUnits;
        shieldPower = s.shieldPower;
    }
    
    // Consultor público para el atributo ammoPower
    
    public float getAmmoPower (){
        
        return ammoPower;
    }
    
    // Consultor público para el atributo fuelUnits
    
    public float getFuelUnits (){
        
        return fuelUnits;
    }
    
    // Consultor público para el atributo shieldPower
    
    public float getShielPower (){
        
        return shieldPower;
    }
    
    /**
     * Método toString
     * @return String que representa el objeto SuppliesPackage
     */
    public String toString() {
        return "Ammo Power: "+ammoPower+", Fuel Units: "+fuelUnits+
                ", Shield Power: "+shieldPower;
    }
    
}
